web/src/add_streams_pill.ts

```diff
@@ -0,0 +1,78 @@
+import * as input_pill from "./input_pill";
+import {set_up} from "./pill_typeahead";
+import * as stream_data from "./stream_data";
+import * as stream_pill from "./stream_pill";
+import type {StreamSubscription} from "./sub_store";
+import type {CombinedPillContainer, CombinedPillItem} from "./typeahead_helper";
+
+type SetUpPillTypeaheadConfig = {
+    pill_widget: CombinedPillContainer;
+    $pill_container: JQuery;
+};
+
+function display_pill(sub: StreamSubscription): string {
+    return "#" + sub.name;
+}
+
+function create_item_from_text(
+    stream_name: string,
+    current_items: CombinedPillItem[],
+): input_pill.InputPillItem<stream_pill.StreamPill> | undefined {
+    stream_name = stream_name.trim();
+    const sub = stream_data.get_sub(stream_name);
+    if (!sub) {
+        return undefined;
+    }
+
+    if (current_items.some((item) => item.type === "stream" && item.stream_id === sub.stream_id)) {
+        return undefined;
+    }
```

Comment:
Why do we need this block of code? L27-L29

Reply:
This code is almost same as `create_item_from_stream_name` function in `stream_pill.ts`. The L27-L29 ensures that duplicate stream pills are not added.

Reply:
Yeah fixed it.